src/auto_archiver/modules/local_storage/local_storage.py

```diff
@@ -20,11 +22,31 @@ def get_cdn_url(self, media: Media) -> str:
     def upload(self, media: Media, **kwargs) -> bool:
         # override parent so that we can use shutil.copy2 and keep metadata
         dest = os.path.join(self.save_to, media.key)
+
+        if len(dest) > self.max_file_length():
+            old_dest_length = len(dest)
+            filename, ext = os.path.splitext(media.key)
+            dir, filename = os.path.split(filename)
+            # see whether we should truncate filename or dir
+            if len(dir) > len(filename):
+                dir = dir[:self.MAX_FILE_LENGTH - len(self.save_to) - len(ext) - len(filename) - 1]
+            else:
+                filename = filename[:self.MAX_FILE_LENGTH - len(self.save_to) - len(ext) - len(filename) - 1]
```

Comment:
Is it the case that the too long error occurs for the totality of size `dir1/dir2/dir3/filename.txt` versus individual parts of the path `dir1TOOLONG/dir2OK/dir2OK/filenameTOOLONG.txt`? this code seems to be addressing the 1st case.

Reply:
So it's a bit complex, but windows basically only supports <260 chars. Mac and Linux support each path 'component' being up to 255 chars, with Mac having no requirement on max total length being, and linux has a MAX_PATH kernel setting which is normally 4096.

In short:
* Windows: 260-character total path limit
* macOS: No enforced total path limit, but 255 char filenames
* Linux: 4096-character total path limit, with 255 char filenames